src/proxy-ui-api/src/main/java/org/niis/xroad/restapi/service/NotificationService.java

```diff
@@ -79,35 +73,35 @@ public AlertStatus getAlerts() {
             alertStatus.setBackupRestoreRunningSince(backupRestoreStartedAt);
             alertStatus.setCurrentTime(OffsetDateTime.now(ZoneOffset.UTC));
         }
-        if (isSecurityServerFullyInitialized()) {
+
+        try {
             alertStatus.setGlobalConfValid(isGlobalConfValid());
-            alertStatus.setSoftTokenPinEntered(isSoftTokenPinEntered());
-        } else {
+            alertStatus.setGlobalConfValidCheckSuccess(true);
+        } catch (CodedException e) {
+            log.error("getting global conf status failed");
             alertStatus.setGlobalConfValid(false);
+            alertStatus.setGlobalConfValidCheckSuccess(false);
+        }
+
+        try {
+            alertStatus.setSoftTokenPinEntered(isSoftTokenPinEntered());
+            alertStatus.setSoftTokenPinEnteredCheckSuccess(true);
+        } catch (Exception e) {
+            log.error("getting soft token pin status failed");
             alertStatus.setSoftTokenPinEntered(false);
+            alertStatus.setSoftTokenPinEnteredCheckSuccess(false);
         }
-        return alertStatus;
-    }
 
-    private boolean isSecurityServerFullyInitialized() {
-        InitializationStatusDto initStatusDto = initializationService.getSecurityServerInitializationStatus();
-        return initStatusDto.isAnchorImported()
-                && initStatusDto.isServerCodeInitialized()
-                && initStatusDto.isServerOwnerInitialized()
-                && initStatusDto.isSoftwareTokenInitialized();
+        return alertStatus;
     }
 
     /**
      * Verifies that the global configuration is valid.
      * @return
      */
     private boolean isGlobalConfValid() {
-        try {
-            globalConfFacade.verifyValidity();
-            return true;
-        } catch (CodedException e) {
-            return false;
-        }
+        globalConfFacade.verifyValidity();
+        return true;
```

Comment:
This method now never returns false, since `globalConfFacade.verifyValidity` either returns true or throws CodedException? 

a `private boolean isGlobalConfValid` that either returns true or throws exception is pretty nonstandard for a Java method, would be better to return true/false in un-exceptional cases. Maybe the old implementation was better after all?

I may have posted misleading comments in the slack thread `I guess the implementation should then inspect the exception to distinguish which case it is` , as it looks like the implementation already did that (well, it did not inspect the error code vs X_OUTDATED_GLOBALCONF, so it was not completely flawless)?

Reply:
Yes, you're right, the old implementation was better. That is now restored. The logic of the global conf check now works so that when the check throws `CodedException` the response is:

```
global_conf_valid: false
global_conf_valid_check_success: true
```

And the the check throws any other exception the response is:

```
global_conf_valid: false
global_conf_valid_check_success: false
```